camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -486,8 +486,8 @@ public DslConfigSupplier(DslComponent component, String keyName) {
             if (targetEntityMaybe.isAbsent()) return Maybe.absent("Target entity not available");
             EntityInternal targetEntity = (EntityInternal) targetEntityMaybe.get();
 
-            ConfigKey<?> key = targetEntity.getEntityType().getConfigKey(keyName);
-            Maybe<? extends Object> result = targetEntity.config().getNonBlocking(key != null ? key : ConfigKeys.newConfigKey(Object.class, keyName));
+            ConfigKey key = targetEntity.getEntityType().getConfigKey(keyName);
```

Comment:
@aledsage I agree in principle.
In this particular case, as Java 8 has a deeper / more precise type inference algorithm, it refuses to compile, because it cannot guarantee that the type of the Maybe (the capture of ?) is actually compatible with both the `key` and the `newConfigKey` call. I wonder how it could be permitted in java 7, actually :-)
I will reinstate the generics with Object as that's the only type that's accepted unequivocally.